v06-either/src/test/java/dev/futa/tutorial/pesel/v06/PeselServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package dev.futa.tutorial.pesel.v06;
+
+import com.google.common.collect.Sets;
+import dev.futa.tutorial.pesel.Gender;
+import dev.futa.tutorial.pesel.PeselDataSetSupplier;
+import io.vavr.control.Either;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static dev.futa.tutorial.pesel.v06.AssertEither.assertEmpty;
+import static dev.futa.tutorial.pesel.v06.AssertEither.assertPresent;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PeselServiceTest {
+  private static PeselDataSetSupplier peselDataSetSupplier;
+  private PeselService peselService;
+
+  @BeforeAll
+  static void beforeAll() {
+    peselDataSetSupplier = new PeselDataSetSupplier();
+  }
+
+  static Stream<Arguments> validPesels() {
+    return peselDataSetSupplier.getValidPeselSet().stream()
+        .map(x -> Arguments.of(x.getPesel(), x.getBirthDate(), x.getGender()));
+  }
+
+  static Set<Arguments> invalidPesels() {
+
+    HashSet<Arguments> arguments =
+        Sets.newHashSet(
+            peselDataSetSupplier.getInvalidLengthPeselSet().stream()
+                .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_LENGTH))
+                .collect(Collectors.toList()));
+
+    arguments.addAll(
+        peselDataSetSupplier.getInvalidCharactersPeselSet().stream()
+            .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_CHARACTERS))
+            .collect(Collectors.toList()));
+
+    arguments.addAll(
+        peselDataSetSupplier.getInvalidChecksumPeselSet().stream()
+            .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_CHECKSUM))
+            .collect(Collectors.toList()));
+
+    arguments.addAll(
+        peselDataSetSupplier.getInvalidEncodedDatePeselSet().stream()
+            .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_DATE))
+            .collect(Collectors.toList()));
+
+    return arguments;
```

Comment:
return Stream
        .of(peselDataSetSupplier.getInvalidLengthPeselSet().stream()
                .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_LENGTH)),
            peselDataSetSupplier.getInvalidCharactersPeselSet().stream()
                .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_CHARACTERS)),
            peselDataSetSupplier.getInvalidChecksumPeselSet().stream()
                .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_CHECKSUM)),
            peselDataSetSupplier.getInvalidEncodedDatePeselSet().stream()
                .map(pesel -> Arguments.of(pesel, FailureReason.INVALID_DATE)))
        .flatMap(Function.identity())
        .collect(Collectors.toSet());
A może coś takiego?